hydromt/components/config.py

```diff
@@ -102,10 +103,9 @@ def write(
     @hydromt_step
     def read(self, path: Optional[str] = None) -> None:
         """Read model config at <root>/{path}."""
-        self._root._assert_read_mode()
         self._initialize(skip_read=True)
         # if path is abs, join will just return path
-        p = path or self._filename
+        p = path or self._default_template_filename or self._filename
         read_path = join(self._root.path, p)
```

Comment:
Can you rechange? We decided with Dirk to add a create method rather than keep reading a template by default for config in write mode for more transparency